packages/modular-station/src/api.ts

```diff
@@ -0,0 +1,224 @@
+import type { AstroConfig, AstroIntegration as NativeIntegration } from 'astro';
+import { definePlugin, type Plugin } from 'astro-integration-kit';
+import { AstroError } from 'astro/errors';
+
+export type AstroIntegration = NativeIntegration;
+
+type HookNames = keyof Required<AstroIntegration['hooks']>;
```

Comment:
are those really hook names? I'd expect suck type to return a union of string but I think it just returns an object?

Reply:
It is a union of string literals for all the names

- `AstroIntegration['hooks']` is the mapping from hook to implementation
- `Required` around that is just to avoid problems with optional properties
- `kyeof` returns only the properties as string literals

Sounds right?

Reply:
my bad I couldn't see `keyof`...